### PR TITLE
fix: Miscellaneous small fixes

### PR DESCRIPTION
**Pull request by arkadyan**
#### Summary of changes
No ticket, just a handful of initial things I came across as I started looking into upgrading Elixir.

- fix: Remove duplicate docs
- fix: This parsing helper shouldn't be a view
- fix: Fix typespec
- fix: Remove function definition that is never reached

---

Before getting review, please check the following:

* [x] Does frontend functionality render and work correctly in IE?
* [x] Have we load-tested any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [x] Are interactive elements accessible to screen readers?
* [x] Have you checked for tech debt you can address in the area you're working in?
* [x] If this change involves routes, does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?
* [x] Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?

### CODE REVIEW

File: apps/site/lib/site_web/views/partial/svg_icon_with_circle.ex [hunk 1]
```diff
@@ -97,7 +97,6 @@ defmodule SiteWeb.PartialView.SvgIconWithCircle do
   defp icon_name(:access), do: "accessible"
   defp icon_name(:no_access), do: "not-accessible"
   defp icon_name(:the_ride), do: "the-ride"
-  defp icon_name(:parking_lot), do: "parking"
```
Comment (amaisano) [inline apps/site/lib/site_web/views/partial/svg_icon_with_circle.ex:100]: Does content know you're removing this icon?

**Review by amaisano — APPROVED** [review]
Generally LGTM, but please check the one comment.

File: apps/site/lib/site_web/views/partial/svg_icon_with_circle.ex (hunk 1 above)
Comment (amaisano) [reply apps/site/lib/site_web/views/partial/svg_icon_with_circle.ex:100]: I think so. The CMS uses this reference: https://github.com/mbta/dotcom/blob/39a294dc4b05e68ea0e32416206640c4f4db62a6/apps/site/lib/site/content_rewriters/liquid_objects.ex#L128

So if that's already switched over we should be good.

Comment (arkadyan) [reply apps/site/lib/site_web/views/partial/svg_icon_with_circle.ex:100]: Nothing is being removed. This is a private function only being called from one place which has a guard on the input. It's not possible for this definition to ever be reached.

**Pull request merged**